corehq/apps/userreports/models.py

```
@@ -331,7 +350,8 @@ def by_domain(cls, domain):
 
     @classmethod
     def by_id(cls, config_id):
-        matching = [ds for ds in cls.all() if ds.get_id == config_id]
-        if not matching:
-            return None
-        return matching[0]
+        for ds in cls.all():
+            if ds.get_id == config_id:
+                return ds
+        raise BadSpecError(_('The data source referenced by this report could '
+                             'not be found.'))
```

this should fail hard, if I understand its usage correctly.